latch/types/metadata.py

```diff
@@ -602,8 +615,49 @@ class SnakemakeMetadata(LatchMetadata):
     """
     A dictionary mapping parameter names to `SnakemakeFileMetadata` objects
     """
+    cores: int = 4
+    """
+    Number of cores to use for Snakemake tasks (equivalent of Snakemake's `--cores` flag)
+    """
+
+    def validate(self):
+        def _contains_none_fields(t: Type, param: Any) -> bool:
+            if is_primitive_type(t) or t in {LatchFile, LatchDir}:
+                return param is None
+
+            if is_list_type(t):
+                if len(get_args(t)) == 0:
+                    raise ValueError(
+                        "Generic Lists are not supported - please specify a subtype,"
+                        " e.g. List[LatchFile]",
+                    )
+                list_typ = get_args(t)[0]
+                for val in param:
+                    if _contains_none_fields(list_typ, val):
+                        return True
+                return False
+
+            assert is_dataclass(t)
+            for field in fields(t):
+                if _contains_none_fields(field.type, getattr(param, field.name)):
+                    return True
+            return False
```

Comment:
TODO: per our conversation, instead of checking null values, do type checking to verify that default parameters match the expected type so that flyte doesn't blow up.